tests/test_containers.py

```diff
@@ -0,0 +1,710 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*
+
+import base64
+import copy
+import json
+import os
+import requests
+import requests_mock
+from requests_mock import NoMockAddress
+import shutil
+import tempfile
+import unittest
+
+import pytest
+import re
+import six
+
+from hamcrest import contains_string
+
+try:
+    from unittest.mock import patch, MagicMock
+except ImportError:  # pragma: no cover
+    from mock import patch, MagicMock
+from urllib3.exceptions import MaxRetryError, ResponseError
+
+from pushsource._impl.utils.containers import (
+    api_version_check,
+    get_manifest,
+    inspect,
+    MT_S2_V1,
+    MT_S2_V1_SIGNED,
+    MT_S2_V2,
+    MT_S2_LIST,
+)
+
+from pushsource._impl.utils.containers.request import (
+    parse_401_response_headers,
+    AuthToken,
+)
+
+
+@pytest.fixture
+def fake_home():
```

Comment:
This function can be simplified by:

- using tmpdir fixture rather than using mkdtemp - then pytest will clean up the directory automatically
- using monkeypatch.setenv - then pytest will undo the env changes automatically

It would look something like:

```python
@pytest.fixture(autouse=True)
def fake_home(tmpdir, monkeypatch):
    home = tmpdir.mkdir("home")
    monkeypatch.setenv("HOME", str(home))
    home.mkdir(".docker").join("config.json").write(...<desired content>)
```

Up to you, but a lot of the current code is reinventing what pytest can already do.